README
delicious-java README


changelog.txt
Version 1.16 - Release Date: Available from CVS - Codename: Dragonforce
-----------------------------------------------------------------------
- Add: Support for "others" attribute

Version 1.15 - Release Date: May 14, 2007 - Codename: Audioslave
---------------------------------------------------------------
- Fix: [ 1718567 ] Preemptive authentication
       https://sourceforge.net/tracker/?func=detail&atid=689673&aid=1718567&group_id=121225
       
Version 1.14 - Release Date: January 18, 2007 - Codename: Minus the Bear
-------------------------------------------------------------------------
- Add: Post:public boolean equals(Object obj)
- Add: public List getAllPosts(String filterTag)
- Fix: [ 1594379 ] wrong scope for variable element in delicious.tld
       https://sourceforge.net/tracker/?func=detail&atid=689673&aid=1594379&group_id=121225

Version 1.13 - Release Date: August 6, 2006 - Codename: Wolfmother
----------------------------------------------------------------------
- Fix: Problems with Non-UTF8-Strings, https://sourceforge.net/tracker/?func=detail&atid=689673&aid=1532303&group_id=121225

Version 1.12 - Release Date: May 17, 2006 - Codename: Loretta Lynn
------------------------------------------------------------------
- HTTPS and API endpoint changes: http://blog.del.icio.us/blog/2006/05/feel_secure.html

Version 1.11 - Release Date: May 8, 2006 - Codename: Ministry
-------------------------------------------------------------
- Fix: Fixed replace property in addPost(...) method where it was not being sent with the request
- Add: Shared property to addPost(...) method to mark a bookmark as shared or not

Version 1.10 - Release Date: February 22, 2006 - Codename: Bob Dylan
--------------------------------------------------------------------
- Fix:    DeliciousUtils UTC date format object now correctly changes dates to UTC instead of local timezone (reported by Erik Thauvin)
- Add:    public void setApiEndpoint(String apiEndpoint)
- Update: Updated Jakarta Commons HttpClient library to 3.0 final.
- Add:    public List getDatesWithPost() method
- Add:    public List getPosts() method
- Add:    public List getRecentPosts() method
- Add:    public boolean addPost(String url, String description) method

Version 1.9 - Codename: Dead Milkmen
------------------------------------
- Add: public List getBundles() method
- Add: public boolean addBundle(String bundleName, String tags) method
- Add: public boolean deleteBundle(String bundleName) method

Version 1.8 - Codename: Beck
----------------------------
- Add: Set method for proxy authentication
- Add: public List getPosts(String filterTag, Date date, String url) method
- Add: public List getPostForURL(String url) method
- Add: public boolean addPost(String url, String description, String extended, String tags, Date date, boolean replace) method
- Add: public List getPostsForTag(String tag) method
- Add: public List getPostsForTags(String[] tags) method

Version 1.7 - Codename: Pet Shop Boys
-------------------------------------
- Add: JSP tag for GetRecentPosts (submitted by Simon Brown)
- Added Simon Brown to project

Version 1.6 - Codename: The Go-Gos
----------------------------------
-    Add: Post.getTagsAsArray(String splitRegex) method
- Update: Set default method retry handler to not retry if HTTP method fails
- Update: Jakarta Commons HTTPClient RC2 library used

Upgrading:
- You may remove the commons-httpclient-3.0-alpha2.jar file and replace it with the commons-httpclient-3.0-rc2.jar.

Version 1.5 - Codename: Porno for Pyros
---------------------------------------
- Update: Changed from GET to POST with addPost method

Version 1.4 - Codename: The Germs
-----------------------------------------
-    Add: DeliciousNotAuthorizedException
- Update: If authorization fails for any of the methods, a DeliciousNotAuthorizedException is thrown
-    Fix: addPost date format fix, https://sourceforge.net/tracker/index.php?func=detail&aid=1088158&group_id=121225&atid=689675 

Version 1.3 - Codename: Eric B. and Rakim
-----------------------------------------
- Add: getLastUpdate() call
- Add: getResultMetaInformation() call to retrieve an object which gets populated on
       certain calls with information about the last call. In the case of calling the
       getAllPosts() method, the result meta information will be populated with a
       java.util.Date object containing the last update time

Version 1.2 - Codename: Deftones
--------------------------------
-    Add: DeliciousException
- Change: API methods may now throw DeliciousException if a SAXException is encounted. IOException is only
          logged as an error.
           
Version 1.1 - Codename: They Might Be Giants
--------------------------------------------
- Add: HTTPClient uses User-Agent string of "del.icio.us Java/(version)" where (version) reflects the
       current version of the library
- Add: Constructor and methods to set HTTP proxy information
- Bug: 1047683 - method Post.getDateAsTime() displays wrong timezone
       https://sourceforge.net/tracker/index.php?func=detail&aid=1047683&group_id=121225&atid=689673

Version 1.0 - Codename: Joy Division
------------------------------------
Initial release

src/del/icio/us/DeliciousConstants.java
/**
 * Copyright (c) 2004-2007, David A. Czarnecki
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * Redistributions of source code must retain the above copyright notice,
 *      this list of conditions and the following disclaimer.
 * Redistributions in binary form must reproduce the above copyright notice,
 *      this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
 * Neither the name of the "David A. Czarnecki" nor the names of
 * its contributors may be used to endorse or promote products derived from
 * this software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
 * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
 * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
 * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
 * EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
 * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
 * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
 * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
 * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
 * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
 * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
 * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
package del.icio.us;

/**
 * DeliciousConstants
 *
 * @author David Czarnecki
 * @version $Id: DeliciousConstants.java,v 1.21 2008/07/30 00:11:22 czarneckid Exp $
 * @since 1.0
 */
public class DeliciousConstants {

    // Default del.icio.us API endpoint
    public static final String API_ENDPOINT = "https://api.del.icio.us/v1/";

    public static final String USER_AGENT_HEADER = "User-Agent";
    public static final String USER_AGENT_VALUE = "del.icio.us Java/1.15";

    // API service endpoints
    public static final String TAGS_GET = "tags/get?";
    public static final String POSTS_DATES = "posts/dates?";
    public static final String POSTS_RECENT = "posts/recent?";
    public static final String POSTS_ALL = "posts/all?";
    public static final String POSTS_GET = "posts/get?";
    public static final String POSTS_ADD = "posts/add?";
    public static final String POSTS_DELETE = "posts/delete?";
    public static final String POSTS_UPDATE = "posts/update?";
    public static final String TAGS_RENAME = "tags/rename?";
    public static final String INBOX_SUBS = "inbox/subs?";
    public static final String INBOX_SUB = "inbox/sub?";
    public static final String INBOX_UNSUB = "inbox/unsub?";
    public static final String INBOX_GET = "inbox/get?";
    public static final String INBOX_DATES = "inbox/dates?";
    public static final String BUNDLES_ALL = "tags/bundles/all?";
    public static final String BUNDLES_SET = "tags/bundles/set?";
    public static final String BUNDLES_DELETE = "tags/bundles/delete?";

    // Various URL parameters for sending data to del.icio.us
    public static final String TAG_PARAMETER = "tag";
    public static final String COUNT_PARAMETER = "count";
    public static final String DT_PARAMETER = "dt";
    public static final String URL_PARAMETER = "url";
    public static final String DESCRIPTION_PARAMETER = "description";
    public static final String EXTENDED_PARAMETER = "extended";
    public static final String TAGS_PARAMETER = "tags";
    public static final String OLD_PARAMETER = "old";
    public static final String NEW_PARAMETER = "new";
    public static final String USER_PARAMETER = "user";
    public static final String REPLACE_PARAMETER = "replace";
    public static final String SHARED_PARAMETER = "shared";
    public static final String NO = "no";
    public static final String YES = "yes"; 

    // Return codes from the service
    public static final String CODE_DONE = "done";
    public static final String CODE_OK = "<ok />";
    public static final String CODE_OK_STANDALONE = "ok";

    // <post .../> tag and attributes
    public static final String POSTS_TAG = "posts";
    public static final String POST_TAG = "post";
    public static final String HREF_ATTRIBUTE = "href";
    public static final String DESCRIPTION_ATTRIBUTE = "description";
    public static final String HASH_ATTRIBUTE = "hash";
    public static final String TAG_ATTRIBUTE = "tag";
    public static final String TIME_ATTRIBUTE = "time";
    public static final String EXTENDED_ATTRIBUTE = "extended";
    public static final String UPDATE_ATTRIBUTE = "update";
    public static final String OTHERS_ATTRIBUTE = "others";

    // <date .../> tag and attributes
    public static final String DATE_TAG = "date";
    public static final String COUNT_ATTRIBUTE = "count";
    public static final String DATE_ATTRIBUTE = "date";

    // <sub .../> tag and attributes
    public static final String SUB_TAG = "sub";
    public static final String USER_ATTRIBUTE = "user";

    // <tag .../> tag and attributes
    public static final String TAG_TAG = "tag";

    // <update .../> tag and attributes
    public static final String UPDATE_TAG = "update";

    // <bundle .../> tag and attributes
    public static final String BUNDLE_TAG = "bundle";
    public static final String BUNDLE_PARAMETER = "bundle";
    public static final String BUNDLE_NAME_ATTRIBUTE = "name";
    public static final String BUNDLE_TAGS_ATTRIBUTE = "tags";

    public static final int DEFAULT_POST_COUNT = 15;
    public static final int MAXIMUM_POST_COUNT = 100;
}